src/library/scanner/recursivescandirectorytask.cpp

```diff
@@ -79,14 +79,11 @@ void RecursiveScanDirectoryTask::run() {
     if (prevHash != newHash) {
         // Rescan that mofo! If importing fails then the scan was cancelled so
         // we return immediately.
-        if (!filesToImport.isEmpty()) {
-            m_pScanner->queueTask(new ImportFilesTask(m_pScanner, m_scannerGlobal,
-                                                      filesToImport, possibleCovers,
-                                                      m_pToken));
-        }
+        m_pScanner->queueTask(new ImportFilesTask(m_pScanner, m_scannerGlobal,
+                                                  dirPath, newHash, prevHashExists,
+                                                  filesToImport, possibleCovers,
+                                                  m_pToken));
 
-        // Insert or update the hash in the database.
-        emit(directoryHashed(dirPath, !prevHashExists, newHash));
```

Comment:
But what actually happens after a directory is hashed is that the files are scheduled for importing in the task queue. If a scan is canceled all pending task are aborted.  This means mixxx thought the files were imported even if the task was canceled